File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int first=-1,end=nums.size()-1,start=0,mid,last=-1;
        while(start<=end){
            mid=end-(end-start)/2;
            if(nums[mid]==target){
                first=mid;
                end=mid-1;
            }
            else if(nums[mid]<target){
                start=mid+1;
            }
            else{
                end=mid-1;
            }

        }
        last=-1,end=nums.size()-1,mid=-1,start=0;
        while(start<=end){
            mid=end-(end-start)/2;
            if(nums[mid]==target){
                last=mid;
                start=mid+1;
            }
            else if(nums[mid]<target){
                start=mid+1;
            }
            else{
                end=mid-1;
            }

        }
        return {first,last};
    }
};

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        result=[]
        def left(nums,target,result):
            start=0
            end=len(nums)-1
            left=-1
            while start<=end:
                mid=int(start+(end-start)/2)
                if nums[mid]==target:
                    left=mid
                    end=mid-1
                elif nums[mid]<target:
                    start=mid+1
                else:
                    end=mid-1
            result.append(left)

        def right(nums,target,result):
            start=0
            end=len(nums)-1
            right=-1
            while start<=end:
                mid=int(start+(end-start)/2)
                if nums[mid]==target:
                    right=mid
                    start=mid+1
                elif nums[mid]<target:
                    start=mid+1
                else:
                    end=mid-1
            result.append(right)
        left(nums,target,result)
        right(nums,target,result)
        return result
        